### PR TITLE
Update `total_supply` when dumping state for the new genesis

### LINKED ISSUE #4665: `state-viewer` fails to dump new genesis state due to mismatching supply
**Describe the bug**
`cargo build --package state-viewer --features nightly_protocol,nightly_protocol_features && ./target/debug/state-viewer --home=~/.near/betanet dump_state`
```
Aug 10 16:24:45.084  INFO near: Opening store database at "/home/user/.near/betanet/data"
Aug 10 16:24:48.135  INFO runtime: Tracking shards: {0}
Generating genesis from state data of #34125348 / Dve5dnycfdsn7F9gWfVUQEwwCD3xqw4SpJ7f9bmqo9Cn
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `1371902924464758940086228587317865`,
 right: `1331261183170775131069856846142903`: wrong total supply', core/chain-configs/src/genesis_validate.rs:85:9
```

This is caused by a recent change forcing validation of total_supply upon each creation of `Genesis`.
What happens is that every epoch the total supply is burned and/or minted. Which leads to a discrepancy between the supply in `records` and in `total_supply` in `genesis.json`.

### PR DESCRIPTION
**Pull request by nikurt**
Fixes #4665

### CODE REVIEW

File: chain/client/tests/chunks_management.rs [hunk 1]
```diff
@@ -100,8 +100,7 @@ fn update_chunk_height_created(
 fn store_partial_encoded_chunk_sanity() {
     init_test_logger();
     let mut env = TestEnv::new(ChainGenesis::test(), 1, 1);
-    let signer =
-        InMemoryValidatorSigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+    let signer = InMemoryValidatorSigner::from_seed("test0", KeyType::ED25519, "test0");
```
Comment (frol) [inline chain/client/tests/chunks_management.rs:103]: Revert this change, it is expected to have `.parse().unwrap()` since we introduced strictly typed AccountId #4621

**Review by frol — APPROVED** [review]

**Review by bowenwang1996 — APPROVED** [review]

File: chain/client/tests/chunks_management.rs (hunk 1 above)
Comment (nikurt) [reply chain/client/tests/chunks_management.rs]: Done.
It took github a few hours to show it as reverted due to a github incident yesterday.
